File: frontend/src/app/core/models/db/Game.ts
export interface Game {
  id: string;
  join_code: string;
  game_state: number[][];
  player_1: Player;
  player_2?: Player;
  current_turn: string;
  status: string;
  moves: Move[];
  created_at: Date;
  winner: string;
}

export interface Player {
  id: string;
  username: string;
  elo_rating: number;
}

export interface Move {
  player: string;
  column: number;
  row: number;
}

File: backend/db/helpers/game.py
from typing import List


def game_helper(game: dict) -> dict:
    return {
        "id": str(game["_id"]),
        "player_1": game["player_1"],
        "player_2": game["player_2"] if game["player_2"] is not None else None,
        "game_state": game["game_state"],
        "status": game["status"],
        "moves": game["moves"],
        "created_at": game["created_at"],
        "join_code": game["join_code"],
        "current_turn": (
            str(game["current_turn"]) if game["current_turn"] is not None else None
        ),
        "winner": game["winner"],
    }


def games_helper(games):
    games_list = []
    for current_game in games:
        game = game_helper(current_game)
        games_list.append(game)
    return games_list
